linkis-public-enhancements/linkis-datasource/linkis-metadata-query/server/src/main/java/org/apache/linkis/metadata/query/server/service/impl/MetadataQueryServiceImpl.java

```diff
@@ -379,32 +381,126 @@ public GenerateSqlInfo getSparkSqlByDsNameAndEnvId(
     DsInfoResponse dsInfoResponse =
         queryDataSourceInfoByNameAndEnvId(dataSourceName, system, userName, envId);
 
-    if (StringUtils.isNotBlank(dsInfoResponse.getDsType())
-        && CacheConfiguration.MYSQL_RELATIONSHIP_LIST
-            .getValue()
-            .contains(dsInfoResponse.getDsType())) {
-      List<MetaColumnInfo> columns =
-          invokeMetaMethod(
-              dsInfoResponse.getDsType(),
-              "getColumns",
-              new Object[] {
-                dsInfoResponse.getCreator(), dsInfoResponse.getParams(), database, table
-              },
-              List.class);
-
-      String sqlConnectUrl =
-          invokeMetaMethod(
-              dsInfoResponse.getDsType(),
-              "getSqlConnectUrl",
-              new Object[] {dsInfoResponse.getCreator(), dsInfoResponse.getParams()},
-              String.class);
-
-      return getSparkSqlByJdbc(database, table, dsInfoResponse.getParams(), columns, sqlConnectUrl);
+    if (StringUtils.isNotBlank(dsInfoResponse.getDsType())) {
+      List<MetaColumnInfo> columns = new ArrayList<>();
+      try {
+        columns =
+            invokeMetaMethod(
+                dsInfoResponse.getDsType(),
+                "getColumns",
+                new Object[] {
+                  dsInfoResponse.getCreator(),
+                  dsInfoResponse.getParams(),
+                  database,
+                  dsInfoResponse.getDsType().equalsIgnoreCase("elasticsearch") ? "_doc" : table
+                },
+                List.class);
+      } catch (Exception e) {
+        logger.warn("Fail to get Sql columns(获取字段列表失败)");
+      }
+      if (CacheConfiguration.MYSQL_RELATIONSHIP_LIST
+          .getValue()
+          .contains(dsInfoResponse.getDsType())) {
+        String sqlConnectUrl =
+            invokeMetaMethod(
+                dsInfoResponse.getDsType(),
+                "getSqlConnectUrl",
+                new Object[] {dsInfoResponse.getCreator(), dsInfoResponse.getParams()},
+                String.class);
+
+        return getSparkSqlByJdbc(
+            database, table, dsInfoResponse.getParams(), columns, sqlConnectUrl);
+      } else if (dsInfoResponse.getDsType().equalsIgnoreCase("kafka")) {
```

Comment:
Is it better to define these as constants or enumerations

Reply:
Ok, thank you for reviewing the code